diffkemp/simpll/SourceCodeUtils.cpp

```diff
@@ -302,3 +315,259 @@ std::vector<SyntaxDifference> findMacroDifferences(
 
     return result;
 }
+
+/// Tries to convert C source syntax of inline ASM (the input may include other
+/// code, the inline asm is found and extracted) to the LLVM syntax.
+/// Returns a pair of strings - the first one contains the converted ASM, the
+/// second one contains (unparsed) arguments.
+std::pair<std::string, std::string> convertInlineAsmToLLVMFormat(
+        std::string input) {
+    int position = input.find("asm");
+
+    if (position == std::string::npos)
+        // No asm present
+        return {"", ""};
+
+    // Skip characters to the first bracket
+    while (position != input.size() && input[position] != '(')
+        ++position;
+
+    // Opening bracket not found
+    if (position == input.size())
+        return {"", ""};
+
+    // Extract the asm body
+    int bracketCounter = 0;
+    std::string extractedBody;
+    do {
+        if (input[position] == '(')
+            ++bracketCounter;
+        else if (input[position] == ')')
+            --bracketCounter;
+        extractedBody += input[position++];
+    } while (position != input.size() && bracketCounter != 0);
+
+    // Closing bracket not found
+    // Note: there is a (currently unfixed) case when the inline asm is split
+    // onto multiple lines and is not joined properly.
+    if (position == input.size())
+        return {"", ""};
+
+    // Remove the first and last bracket from the expression
+    extractedBody.erase(0, 1);
+    extractedBody.erase(extractedBody.size() - 2, 1);
+
+    // Do section joining
+    std::string newBody;
+    std::string arguments;
+    bool insideSection = false, argumentsFlag = false;
+    position = 0;
+    while (position < extractedBody.size()) {
```

Comment:
Split into two parts (before ":" and after).
Implement the part before ":" using string manipulation functions.